hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRequestFutureImpl.java

```diff
@@ -715,17 +715,24 @@ private void receiveGlobalFailure(MultiAction rsActions, ServerName server, int
     errorsByServer.reportServerError(server);
     Retry canRetry = errorsByServer.canTryMore(numAttempt) ? Retry.YES : Retry.NO_RETRIES_EXHAUSTED;
 
-    cleanServerCache(server, t);
+    // Do not update cache if exception is from failing to submit action to thread pool
+    if (!(t instanceof RejectedExecutionException)) {
```

Comment:
this receiveGlobalFailure method is private and only called in 3 places. Rather than add another instanceof check here, thoughts on just adding a boolean argument to the method `shouldClearCache`? We can pass in false everywhere, except in the place where we already `instanceof RejectedExecutionException` in sendMultiAction